metabase/metabase
# allow re-ordering and re-naming pie chart slices

Closes https://github.com/metabase/metabase/issues/9312
Closes https://github.com/metabase/metabase/issues/37145

### Description

Allows users to rename and reorder pie chart slices. This is achieved by adding two new viz settings, `pie.rows` and `pie.sort_rows`.

`pie.rows` is the source of truth for slice order, name, and color. `pie.sort_rows` is a boolean flag determining whether the slices should automatically be sorted in descending order by measure value. If the user ever re-orders the slices manually from the sidebar, this will be set to false, and their manual order will be respected.

Regarding the behavior when rows are added and removed from the query results, I've recorded a loom explaining that: https://www.loom.com/share/03cee849cfae46128c5d46e9a45e9d30?sid=de71a64b-a6f7-45cc-9af3-26a776b9e01c

### How to verify

1. Create a pie chart
2. Try reordering, renaming, and recoloring slices
3. Add more rows to query result (can do this through limiting rows)
4. Confirm slices for these rows were added to end of chart sensibly
5. Remove rows (e.g. through limiting), then re-add them. Confirm their settings were preserved.

### Demo

^ see loom linked above

### Checklist
